common/elasticsearch/aws.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package elasticsearch
+
+import (
+	"fmt"
+	awsauth "github.com/smartystreets/go-aws-auth"
+	"net/http"
+	"os"
+)
+
+type AWSSigningTransport struct {
+	HTTPClient  *http.Client
+	Credentials awsauth.Credentials
+}
+
+// RoundTrip implementation
+func (a AWSSigningTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return a.HTTPClient.Do(awsauth.Sign4(req, a.Credentials))
+}
+
+func createAWSClient() (*http.Client, error) {
+	id := os.Getenv("AWS_ACCESS_KEY_ID")
```

Comment:
I'm still not sure what the difference between exposing it via ServiceAccount's secret, and a regular secret.. also I'm not sure how I access the ServiceAccount's secret.. via reading the file `/var/run/secrets/kubernetes.io/serviceaccount/aws_access_key`?*

Also, these ENVs are the ["common practice" by AWS GO SDK](https://github.com/aws/aws-sdk-go/wiki/configuring-sdk#environment-variables), I tried to align with them here(both with names and recommendations)

*The documentation of k8s for SA is pretty lack of info..